### DISCUSSION TITLE
Buildroot + PB = love?

### DISCUSSION THREAD

**Discussion by phil-flip** [pocketbase/pocketbase#1543] [Q&A]
tl;dr: Have there been made any efforts to compile PB with Buildroot and run it for embedded software.

Hello PB-Community,
I'm fairly new to frontend development and got straight dragged into making a Management-Website for a portable product. Since I dealt with some IoT stuff before, I was planning to use Buildroot to create the entire firmware. Since PB is already precompiled into one binary I got intrigued, if I can get away of using it as my backend for the SvelteKit project I have been working on. I already saw that there are ARM-based versions available, but my main question is, if there are _any_ dependencies, that it possibly needs. Buildroot basically strips the OS down to its essentials and making it very fast.

The main problem with Buildroot is not very common at all, and therefor there are not many resources about it. So I wanted to do the first step and see if there have been already tries to get it working.

Before, some people get any headaches from this crazy approach by banging their hands against harder material: I'm kinda new to both, and it seems to be a simple and valid approach. But incase there is any better approach to deal with Frontends and their data on low-level devices, let me know, before my head starts to hurt too. ^^'

Regards, Phil

**Comment by ganigeorgiev** [comment]
I have no experience with Buildroot and I'm not sure how exactly it works.
Once compiled, PocketBase doesn't really have any external dependency because the result binary would be statically linked. This is the case when `CGO_ENABLED=0` (all prebuilt binaries are generated with disabled cgo) and it should compile fine on the following platforms:

```
darwin  amd64
darwin  arm64
freebsd amd64
freebsd arm64
linux   386
linux   amd64
linux   arm
linux   arm64
linux   ppc64le
linux   riscv64
windows amd64
windows arm64
```

If your targetted platform is missing from the prebuilt binaries in the ["Assets" releases section](https://github.com/pocketbase/pocketbase/releases), you can prebuild PocketBase on your own by:

0. [Install Go](https://go.dev/doc/install) (if not already)
1. Clone/download the PocketBase repo
2. Navigate to `examples/base`
3. Run `GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build`
   _(https://go.dev/doc/install/source#environment)_

**Reply by phil-flip** [reply]
Thank you for the detailed info about the images and build process. I will try it out, when it's time ^^
